spaceship/lib/spaceship/connect_api/users/users.rb

```diff
@@ -28,6 +28,36 @@ def delete_user(user_id: nil)
           users_request_client.delete("users/#{user_id}")
         end
 
+        # Update existing user
+        def patch_user(user_id:, all_apps_visible:, provisioning_allowed:, roles:, visible_app_ids:)
+          body = {
+            data: {
+              type: 'users',
+              id: user_id,
+              attributes: {
+                allAppsVisible: all_apps_visible,
+                provisioningAllowed: provisioning_allowed,
+                roles: roles
+              },
+              relationships: {
+                visibleApps: {
+                  data: visible_app_ids.map do |app_id|
+                    {
+                      type: "apps",
+                      id: app_id
+                    }
+                  end
+                }
+              }
+            }
+          }
+
+          # Avoid API error: You cannot set visible apps for this user because the user's roles give them access to all apps.
+          body[:data].delete(:relationships) if all_apps_visible
+
+          users_request_client.patch("users/#{user_id}", body)
+        end
+
         # Add app permissions for user
         # @deprecated Use {#post_user_visible_apps} instead.
         def add_user_visible_apps(user_id: nil, app_ids: nil)
```

Comment:
Couldn't this function be deprecated now, since we can use the `patch users/id` for everything, including adding visible apps to a given user? 🤔 

Same for `post_user_visible_apps`, `patch_user_visible_apps`, `delete_user_visible_apps`…? Or am I misunderstanding their purpose? (I don't understand why there's post and patch user visible apps 🤯

Reply:
idk, there are API endpoints that are designed to do precisely this: [POST to Add Apps](https://developer.apple.com/documentation/appstoreconnectapi/add_visible_apps_to_a_user), [PATCH to Replace Apps](https://developer.apple.com/documentation/appstoreconnectapi/replace_the_list_of_visible_apps_for_a_user) and [DELETE to Remove Apps](https://developer.apple.com/documentation/appstoreconnectapi/remove_visible_apps_from_a_user). All four API calls are available from v1.0+, and none of them is deprecated yet. 
I agree that this makes those functions look a little bit weird but unlike the whole User PATCH endpoint, they are designed to work only with the apps list.